diagnostics/physics/sst_trends.py

```diff
@@ -25,101 +28,132 @@ def get_3d_trends(x, y):
     return trends
 
 
-def plot_sst_trends(pp_root, label):
+def plot_sst_trends(pp_root, label, config):
     model = (
         open_var(pp_root, 'ocean_monthly', 'tos')
-        .sel(time=slice('2005', '2019'))
+        .sel(time=slice(config['start_year'], config['end_year']))
         .resample(time='1AS')
         .mean('time')
         .load()
     )
-    model_grid = xarray.open_dataset('../data/geography/ocean_static.nc')
+    logger.info("MODEL: %s",model)
+    model_grid = xarray.open_dataset( config['model_grid'])
+    logger.info("MODEL_GRID: %s",model_grid)
+
+    # Verify that xh/yh are set as coordinates, then make sure model coordinates match grid data
+    model_grid = model_grid.assign_coords( {'xh':model_grid.xh, 'yh':model_grid.yh } )
+    model = xarray.align(model_grid, model, join='override', exclude='time')[1]
+    logger.info("Successfully modified coordinates of model grid, and aligned model coordinates to grid coordinates")
 
     model_trend = get_3d_trends(model['time.year'], model) * 10 # -> C / decade
+    # Convert to Data Array, since xskillscore expects dataarrays to calculate skill metrics
     model_trend = xarray.DataArray(model_trend, dims=['yh', 'xh'], coords={'yh': model.yh, 'xh': model.xh})
+    logger.info("MODEL_TREND: %s", model_trend)
 
-    oisst = (
-        xarray.open_mfdataset([f'/work/acr/oisstv2/sst.month.mean.{y}.nc' for y in range(2005, 2020)])
-        .sst
-        .sel(lat=slice(0, 60), lon=slice(360-100, 360-30))
-        .resample(time='1AS')
-        .mean('time')
-        .load()
-    )
-    oisst_trend = get_3d_trends(oisst['time.year'], oisst) * 10 # -> C / decade
+    target_grid = model_grid[ config['rename_map'].keys() ].rename( config['rename_map'] )
 
-    glorys = (
-        xarray.open_dataset('/work/acr/mom6/diagnostics/glorys/glorys_sfc.nc')
-        ['thetao']
-        .sel(time=slice('2005', '2019'))
-        .resample(time='1AS')
-        .mean('time')
-    )
+    # Process OISST and get trend
+    mom_to_oisst, oisst, oisst_lonc, oisst_latc = process_oisst(config, target_grid, model, start =  int(config['start_year']),
+                                                                end = int(config['end_year'])+1, resamp_freq = '1AS',
+                                                                do_regrid = False) # (note that model data is not used if do_regrid = False)
```

Comment:
Since we're following Andrew's approach to interpolate the model to OISST and GLORYS to the model, we might want to remove the `do_regrid` option in both `process_oisst` and `process_glorys`. This way, the scripts will just perform the intended interpolations (model to OISST, GLORYS to model), and we can clean up any redundant code (e.g., line #64).

Reply:
I'm just now taking a look at this, and I'm not sure that we can remove the `do_regrid` option from `process_glorys`. This script regrids the `glorys_trend`  to the model grid, and `glorys_trend` is calculated from the `glorys_ave` dataset that is created and returned by this function. 

It looks like the only options are to either 1.) call the `get_3d_trends` function within `process_glorys` which would require adding a command line flag to specify when this function is needed, or 2.) Maintaining the current design that gives you the option to either return the regridder or the regridded dataset. Let me know which sounds better, or if I'm missing another option.

Reply:
@uwagura, I personally prefer the first option of adding an optional flag in `process_glorys` to allow the function to return trend output when needed, as I think the `do_regrid` option could cause some confusion (since we already know regridding is always required in our case). However, it’s entirely up to you!